### PR TITLE
ccmlib: Remove usages of distutils

### LINKED ISSUE #537: Fix compatibility with Python 3.12
In https://github.com/scylladb/java-driver we run scylla-ccm for integration tests in GitHub Actions. Recently, GitHub Actions switched to using Python 3.12 by default and CCM fails to start with the following error:

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.12.0/x64/bin/ccm", line 7, in <module>
    from ccmlib.cmds import command, cluster_cmds, node_cmds
  File "/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/ccmlib/cmds/command.py", line 6, in <module>
    from ccmlib.cluster_factory import ClusterFactory
  File "/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/ccmlib/cluster_factory.py", line 5, in <module>
    from ccmlib import common, repository
  File "/opt/hostedtoolcache/Python/3.12.0/x64/lib/python3.12/site-packages/ccmlib/repository.py", line 15, in <module>
    from distutils.version import LooseVersion
ModuleNotFoundError: No module named 'distutils'
```

`distutils` was removed in Python 3.12.

### PR DESCRIPTION
**Pull request by k0machi**
This change removes all uses of distutils modules and replaces them with
either direct replacements or extractions from the library.
LooseVersion is extracted into ccmlib.utils.version and dir_util.copy_tree
is replaced by shutil.copytree.

Fixes #537

### CODE REVIEW

File: tests/test_config.py [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+    normalized_val = val.lower()
+    if normalized_val in ('y', 'yes', 't', 'true', 'on', '1'):
+        return True
+    if normalized_val in ('n', 'no', 'f', 'false', 'off', '0'):
+        return False
+    else:
+        raise ValueError(f"invalid truth value {val}")
+
+
+DEV_MODE = strtobool(os.environ.get("DEV_MODE", "False"))
```
Comment (fruch) [inline tests/test_config.py:14]: DEV_MODE isn't used, lets remove it, instead of introducing new code

**Review by fruch — CHANGES_REQUESTED** [review]
Lets avoid dragging in deprecated code

File: ccmlib/repository.py [hunk 2]
```diff
@@ -385,15 +386,15 @@ def get_tagged_version_numbers(series='stable'):
     for ref in (i.get('ref', '') for i in json.loads(tag_url.read())):
         m = tag_regex.match(ref)
         if m:
-            releases.append(LooseVersion(m.groups()[0]))
+            releases.append(Version(m.groups()[0]))
```
[the remark this answers is not in the document]
Comment (k0machi) [reply ccmlib/repository.py:389]: Will do

File: tests/test_config.py (hunk 1 above)
Comment (k0machi) [reply tests/test_config.py:14]: Will do

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**